### PR TITLE
Track message sequences for Waku dogfooding

### PR DESCRIPTION
**Pull request by adklempner**
Adds a table and endpoint for tracking Waku data for the dogfooding app

This shouldn't effect any status-related telemetry

### CODE REVIEW

File: telemetry/sql/000008_waku_push_filter.up.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+CREATE TABLE IF NOT EXISTS wakuPushFilter (
+	id SERIAL PRIMARY KEY,
+	walletAddress VARCHAR(255),
+	peerIdSender VARCHAR(255) NOT NULL,
+	peerIdReporter VARCHAR(255) NOT NULL,
```
Comment (weboko) [inline telemetry/sql/000008_waku_push_filter.up.sql:5]: what is `reporter` in this context? is it the node that we sent message to?

File: telemetry/waku_metrics.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+type WakuTelemetryRequest struct {
+	Id            int               `json:"id"`
+	TelemetryType WakuTelemetryType `json:"telemetryType"`
+	TelemetryData *json.RawMessage  `json:"telemetryData"`
+}
+
+type TelemetryPushFilter struct {
+	ID             int    `json:"id"`
+	WalletAddress  string `json:"walletAddress"`
```
Comment (weboko) [inline telemetry/waku_metrics.go:23]: seems we don't need it for dogfooding

File: telemetry/sql/000007_error_sending_envelope.up.sql [hunk 3]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE IF NOT EXISTS errorSendingEnvelope (
```
Comment (weboko) [inline telemetry/sql/000007_error_sending_envelope.up.sql:1]: why do we need this table? it doesn't seem needed for dogfooding and maybe can be combined with other envelope tables?

File: telemetry/server.go [hunk 4]
```diff
@@ -349,7 +358,65 @@ func (s *Server) rateLimit(next http.Handler) http.Handler {
 	})
 }
 
+func (s *Server) createWakuTelemetry(w http.ResponseWriter, r *http.Request) {
```
Comment (weboko) [inline telemetry/server.go:361]: I like how it sounds :D

File: telemetry/sql/000008_waku_push_filter.up.sql (hunk 1 above)
Comment (adklempner) [reply telemetry/sql/000008_waku_push_filter.up.sql]: Reporter is the node being run by the same user who sends this data to telemetry. When sending a message via lightpush, reporter and sender will be the same.

File: telemetry/waku_metrics.go (hunk 2 above)
Comment (adklempner) [reply telemetry/waku_metrics.go:23]: It's an optional field in the db table

File: telemetry/sql/000007_error_sending_envelope.up.sql (hunk 3 above)
Comment (adklempner) [reply telemetry/sql/000007_error_sending_envelope.up.sql]: removed as it will be handled in separate PR after this one

**Review by weboko — APPROVED** [review]
great! let's iterate in following PRs

File: telemetry/waku_metrics.go [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	Timestamp      int64  `json:"timestamp"`
+	CreatedAt      int64  `json:"createdAt"`
+}
+
+func (r *TelemetryPushFilter) put(db *sql.DB) error {
+	stmt, err := db.Prepare("INSERT INTO wakuPushFilter (peerIdSender, peerIdReporter, sequenceHash, sequenceTotal, sequenceIndex, contentTopic, pubsubTopic, timestamp, createdAt) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9) RETURNING id;")
+	if err != nil {
+		return err
+	}
+
```
Comment (richard-ramos) [inline telemetry/waku_metrics.go:40]:
```suggestion
defer stmt.Close()
```

File: telemetry/server.go [hunk 6]
```diff
@@ -349,7 +350,59 @@ func (s *Server) rateLimit(next http.Handler) http.Handler {
...
+	start := time.Now()
+	var telemetryData []WakuTelemetryRequest
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&telemetryData); err != nil {
+		log.Println(err)
+		http.Error(w, "Failed to decode telemetry data", http.StatusBadRequest)
+		return
+	}
+
+	var errorDetails []map[string]interface{}
```
Comment (richard-ramos) [inline telemetry/server.go:363]: Wouldn't it be better to create a type instead?
```
type ErrorDetail struct {
  Id string `json:"Id"`
  Error string `json:"Error"`
}
```
And then use that type in 
```
var errorDetails []map[string]ErrorDetail
```

**Review by richard-ramos — APPROVED** [review]

File: telemetry/waku_metrics.go (hunk 5 above)
Comment (adklempner) [reply telemetry/waku_metrics.go:40]: added

File: telemetry/server.go (hunk 6 above)
Comment (adklempner) [reply telemetry/server.go]: added and included in http response when errors are encountered

**Pull request merged**